model/src/pyrenew/latent/hospitaladmissions.py

```diff
@@ -184,13 +211,33 @@ def sample(
             mode="full",
         )[: infection_hosp_rate_t.shape[0]]
 
-        # Applying the day of the week effect
-        latent_hospital_admissions = (
-            latent_hospital_admissions
-            * self.day_of_week_effect_rv(
-                n_timepoints=latent_hospital_admissions.size, **kwargs
-            )[0].value
-        )
+        # Applying the day of the week effect. For this we need to:
+        # 1. Get the day of the week effect
+        # 2. Identify the offset of the latent_infections
+        # 3. Apply the day of the week effect to the latent_hospital_admissions
+        dow_effect_sampled = self.day_of_week_effect_rv(**kwargs)[0]
+
+        if dow_effect_sampled.value.size != 7:
+            raise ValueError(
+                "Day of the week effect should have 7 values. "
+                f"Got {dow_effect_sampled.value.size} instead."
+            )
+
+        # Identifying the offset
+        inf_offset = (
+            latent_infections.t_start % 7
+            if latent_infections.t_start is not None
+            else 0
+        ) + self.obs_data_first_day_of_the_week
```

Comment:
Not sure how to think of this… is this saying that we assume `t=0` to be a Monday?

Reply:
Not assuming. It is part of the design. See the documentation.

Reply:
Also, worth noticing `datetime` and `numpy.datetime64` both use Monday as anchor.

Reply:
Can you explain (either in GitHub or a code comment) the logic behind these lines? I find it hard to follow.

Reply:
It defines the offset in terms of days between 0-6, regardless of the starting date of the data. If the seeding process starts -4 days, -4 % 7 = 3 that would be Sunday->Saturday->Friday->Thursday. Otherwise, if it is 4 then 4 % 7 = 4, thus Friday. This would work with bigger numbers too. For instance, adding three weeks, we would have -25 % 7 = 3.